qcodes/instrument/parameter.py

```diff
@@ -288,8 +291,21 @@ def __init__(
         snapshot_exclude: bool = False,
         max_val_age: Optional[float] = None,
         vals: Optional[Validator[Any]] = None,
-        abstract: Optional[bool] = False
+        abstract: Optional[bool] = False,
+        bind_to_instrument: bool = True,
     ) -> None:
+
+        if instrument is not None and bind_to_instrument:
+            existing_parameter = instrument.parameters.get(name, None)
+
+            if existing_parameter:
+
+                if not existing_parameter.abstract:
+                    raise KeyError(
+                        f"Duplicate parameter name {name} on instrument {instrument}"
+                    )
+
+            instrument.parameters[name] = self
```

Comment:
shall we do this after ``if not str(name).isidentifier():`` check? also after the __init__ of the parent class (metadatable)? or would it break something?